Automattic/wp-calypso
# JP Pro Dashboard: Add the license lightbox in the Issue License screen.

> This PR is part of a 2-part PR that aims to bring the Product Lightbox to the Jetpack agency Pro Dashboard - Issue License screen. The first PR can be located here: #76760

This PR updates the Issue License screen to display the product information when pressing the more about links in the product or bundle card. 

<img width="1100" alt="Screen Shot 2023-05-29 at 3 56 18 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/098c09c6-4ac0-4351-8fe3-402484857668">


<img width="1109" alt="Screen Shot 2023-05-29 at 3 55 26 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/8ad672bf-8261-4826-b3ee-1ee0064a17c6">


Additionally, it was planned to reuse the existing product lightbox component on the pricing page. However, due to the component's complexity and how tightly coupled it is to the checkout flow, I decided it would be better to create a separate lightbox to reduce the risk of breaking something on the pricing page if we try to tweak it for the agency's needs.

In this PR, I created a dumb version of the product lightbox that is much more reusable than the previous one. I also moved the component to the shared component folder, given that this will be used on multiple codebases.

I also plan to update the product lightbox on the pricing page to use the new lightbox on a separate PR to avoid bloating the size of this already huge PR.

Related to 1202619025189113-as-1203886504803361

## Proposed Changes

*  Add a dumb version that can be reused for the pricing page and the agency dashboard. The new lightbox component should also allow flexibility on what content to render. 
* Add a `JetpackProductInfo` component that will render product info. The idea is, later, when we decide on new Product information specific to the agency dashboard, we can now easily swap it with a different component. I plan to reuse this component for the product lightbox on the pricing page on a separate PR.
* Implement the `LicenseLightbox` component that utilizes the new reusable components.

## Testing Instructions
* Make sure you have an agency account before testing this.
* Use the Jetpack cloud live link below and append `/partner-portal/issue-license.`
* In the issue license screen. Click any 'more about' link and confirm the lightbox shows up.

## Pre-merge Checklist



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1203886504803361
